worldedit-core/src/main/java/com/sk89q/worldedit/EditSession.java

```diff
@@ -1478,29 +1488,45 @@ public int makeSphere(Vector pos, Pattern block, double radiusX, double radiusY,
                         }
                     }
 
-                    if (setBlock(pos.add(x, y, z), block)) {
-                        ++affected;
-                    }
-                    if (setBlock(pos.add(-x, y, z), block)) {
-                        ++affected;
-                    }
-                    if (setBlock(pos.add(x, -y, z), block)) {
-                        ++affected;
-                    }
-                    if (setBlock(pos.add(x, y, -z), block)) {
-                        ++affected;
-                    }
-                    if (setBlock(pos.add(-x, -y, z), block)) {
-                        ++affected;
-                    }
-                    if (setBlock(pos.add(x, -y, -z), block)) {
-                        ++affected;
-                    }
-                    if (setBlock(pos.add(-x, y, -z), block)) {
-                        ++affected;
+                    if (directionVector != null) {
+                        Vector vectors[] = {pos.add(x, y, z), pos.add(-x, y, z), pos.add(x, y, -z), pos.add(-x, y, -z), pos.add(x, -y, z), pos.add(-x, -y, z), pos.add(x, -y, -z), pos.add(-x, -y, -z)};
```

Comment:
Storing everything in a temporary array becomes superfluous if you add a function that does the filtering.
@sk89q do we have a way to add temporary filters to whatever is behind setBlock already?